File: sdk/go/get_cert.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

type certGetRequest struct {
	Domains   string `json:"domains,omitempty"`
	CertID    int64  `json:"certId,omitempty"`
	AutoApply bool   `json:"autoApply"`
	Format    string `json:"format,omitempty"`
}

func requireEnv(name string) (string, error) {
	value := os.Getenv(name)
	if value == "" {
		return "", fmt.Errorf("missing environment variable: %s", name)
	}
	return value, nil
}

func boolEnv(name string, defaultValue bool) bool {
	value := os.Getenv(name)
	if value == "" {
		return defaultValue
	}
	switch strings.ToLower(value) {
	case "1", "true", "yes", "y":
		return true
	default:
		return false
	}
}

func main() {
	if err := run(); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
}

func run() error {
	keyID, err := requireEnv("CERTD_KEY_ID")
	if err != nil {
		return err
	}
	keySecret, err := requireEnv("CERTD_KEY_SECRET")
	if err != nil {
		return err
	}

	client, err := NewCertdClient(keyID, keySecret)
	if err != nil {
		return err
	}
	if baseURL := os.Getenv("CERTD_BASE_URL"); baseURL != "" {
		client.BaseURL = baseURL
	}
	client.Encrypt = boolEnv("CERTD_ENCRYPT", false)

	reqBody := certGetRequest{
		Domains:   os.Getenv("CERTD_DOMAINS"),
		AutoApply: boolEnv("CERTD_AUTO_APPLY", false),
		Format:    os.Getenv("CERTD_FORMAT"),
	}
	if certID := os.Getenv("CERTD_CERT_ID"); certID != "" {
		reqBody.CertID, err = strconv.ParseInt(certID, 10, 64)
		if err != nil || reqBody.CertID <= 0 {
			return fmt.Errorf("CERTD_CERT_ID must be a positive integer")
		}
	}
	if reqBody.CertID == 0 && reqBody.Domains == "" {
		return fmt.Errorf("set CERTD_CERT_ID or CERTD_DOMAINS")
	}

	respBody, err := client.GetCert(reqBody)
	if err != nil {
		return err
	}
	fmt.Println(string(respBody))
	return nil
}

File: sdk/php/get_cert.php
<?php

require_once __DIR__ . '/CertdClient.php';

function require_env(string $name): string
{
    $value = getenv($name);
    if ($value === false || $value === '') {
        throw new RuntimeException("Missing environment variable: {$name}");
    }
    return $value;
}

function bool_env(string $name, bool $default = false): bool
{
    $value = getenv($name);
    if ($value === false || $value === '') {
        return $default;
    }
    return in_array(strtolower($value), ['1', 'true', 'yes', 'y'], true);
}

try {
    $client = new CertdClient(require_env('CERTD_KEY_ID'), require_env('CERTD_KEY_SECRET'), [
        'baseUrl' => getenv('CERTD_BASE_URL') ?: 'http://127.0.0.1:7001',
        'encrypt' => bool_env('CERTD_ENCRYPT'),
    ]);

    $params = [
        'autoApply' => bool_env('CERTD_AUTO_APPLY'),
    ];
    if (getenv('CERTD_CERT_ID')) {
        if (!ctype_digit(getenv('CERTD_CERT_ID'))) {
            throw new RuntimeException('CERTD_CERT_ID must be a positive integer');
        }
        $params['certId'] = intval(getenv('CERTD_CERT_ID'));
    }
    if (getenv('CERTD_DOMAINS')) {
        $params['domains'] = getenv('CERTD_DOMAINS');
    }
    if (getenv('CERTD_FORMAT')) {
        $params['format'] = getenv('CERTD_FORMAT');
    }
    if (empty($params['certId']) && empty($params['domains'])) {
        throw new RuntimeException('Set CERTD_CERT_ID or CERTD_DOMAINS');
    }

    echo $client->getCert($params) . PHP_EOL;
} catch (Throwable $e) {
    fwrite(STDERR, $e->getMessage() . PHP_EOL);
    exit(1);
}
